apps/dialer/js/dialer.js

```diff
@@ -0,0 +1,599 @@
+        function(result) {
+          if (result && (typeof result === 'object') && result.number) {
+            LazyLoader.load(['../shared/js/sim_settings_helper.js'], function() {
+              SimSettingsHelper.getCardIndexFrom('outgoingCall', function(ci) {
+                // If the default outgoing call SIM is set to "Always ask", or
+                // is unset, we place this call on the SIM that was used the
+                // last time we were in a phone call with this number/contact.
+                if (ci === undefined || ci === null ||
+                    ci == SimSettingsHelper.ALWAYS_ASK_OPTION_VALUE) {
+                  ci = result.serviceId;
+                }
+
+                CallHandler.call(result.number, ci);
+              });
+            });
+          } else {
+            console.log('Could not get the group at: ' + position +
+                        '. Error: ' + result);
+          }
+        });
+      }
+
+      return;
+    }
+
+    // Other commands needs to be handled from the call screen
+    if (callScreenWindowReady) {
+      sendCommandToCallScreen('BT', command);
+    } else {
+      // We queue the commands while the call screen is loading
+      btCommandsToForward.push(command);
+    }
+  }
+
+  /* === Headset Support === */
+  function headsetCommandHandler(message) {
+    sendCommandToCallScreen('HS', message);
+  }
+
+  /*
+    Send commands to the callScreen via post message.
+    @type: Handler to be used in the CallHandler. Currently managing to
+           kind of commands:
+           'BT': bluetooth
+           'HS': headset
+           '*' : for general cases, not specific to hardware control
+    @command: The specific message to each kind of type
+  */
+  function sendCommandToCallScreen(type, command) {
+    if (!callScreenWindow) {
+      return;
+    }
+
+    var message = {
+      type: type,
+      command: command
+    };
+    callScreenWindow.postMessage(message, '*');
+  }
+
+  /* === postMessage support === */
+
+  // Receiving messages from the callscreen via post message
+  //   - when the call screen is closing
+  //   - when the call screen is ready to receive messages
+  //   - when we need to hide or show navbar
+  function handleMessage(evt) {
+    if (evt.origin !== COMMS_APP_ORIGIN) {
+      return;
+    }
+
+    var data = evt.data;
+
+    if (data === 'closing') {
+      handleCallScreenClosing();
+    } else if (data === 'ready') {
+      handleCallScreenReady();
+    } else if (!data.type) {
+      return;
+    } else if (data.type === 'contactsiframe') {
+      handleContactsIframeRequest(data.message);
+    } else if (data.type === 'hide-navbar') {
+      NavbarManager.hide();
+    } else if (data.type === 'show-navbar') {
+      NavbarManager.show();
+    }
+  }
+
+  window.addEventListener('message', handleMessage);
+
+  /* === Calls === */
+  function call(number, cardIndex) {
+    if (engineeringModeKey && number === engineeringModeKey) {
```

Comment:
You can kill the engineering mode key support and all the associated tests if you feel like it: I've personally removed that API from gecko :-)